### ISSUE TITLE
Valid license? 

### ISSUE THREAD

**Issue by jviramo1** [Greenzie/boustrophedon_planner#4]
We would like to contribute to the project but we're concerned about the licensing. 
The repo is setup as "MIT" but you appear to be using the CGAL library which is either GPL or LGPL so we have concerns about the licensing restrictions attached to this code. 
Have you confirmed that you're allowed to release this code as "MIT"?

**Comment by seebq** [comment]
Now I’m concerned too! Thanks for bringing this to our attention.

I thought we were using only the kernel and support but let us check
further.

How would we confirm other than going through and validating based on
cgal’s requirements?

**Comment by jviramo1** [comment]
The confusing part is that there's no single license for CGAL, each package seems to be licensed individually. So I think the first step is understanding what exactly you're using from CGAL and verifying the license. 
If it's LGPL then you may be able to make it work as a linked library? 
If what you're using is GPL then my understanding is that anything that touches that code needs to be released as GPL.

**Comment by seebq** [comment]
Yeah we talked internally today and we think the right answer is to talk to an IP lawyer knowledgeable in open source software. 

There is a fine line with LGPL and GPL, and the GPL is "infectious," ("you are furthermore obliged to distribute the source code of your own software under the GPL"), but as to whether this ROS node we released, which is the entire node, may need to be LGPL or GPL now, and how far up it goes if you decide to use this in your "application" via network connections, or compile it altogether in your build process....

Regardless, we'll find out more and get back to you.  Thanks for bringing this to our attention.
